tests/Microsoft.Identity.Test.Unit/ManagedIdentityTests/CloudShellTests.cs

```diff
@@ -27,65 +25,19 @@ public void TestInitialize()
-        [DataRow(Endpoint, "https://management.azure.com")]
-        [DataRow(Endpoint, "https://management.azure.com/.default")]
-        public async Task CloudShellHappyPathAsync(
-            string endpoint, 
-            string scope)
+        [DataRow(TestConstants.ClientId, UserAssignedIdentityId.ClientId)]
+        [DataRow("resourceId", UserAssignedIdentityId.ResourceId)]
+        public async Task CloudShellUserAssignedManagedIdentityNotSupportedAsync(string userAssignedClientId, UserAssignedIdentityId userAssignedIdentityId)
         {
             using (new EnvVariableContext())
             using (var httpManager = new MockHttpManager())
             {
-                SetEnvironmentVariables(endpoint);
+                ManagedIdentityTests.SetEnvironmentVariables(ManagedIdentitySourceType.CloudShell, ManagedIdentityTests.CloudShellEndpoint);
 
-                IConfidentialClientApplication cca = ConfidentialClientApplicationBuilder
-                    .Create("clientId")
-                    .WithHttpManager(httpManager)
-                    .WithExperimentalFeatures()
-                    .Build();
-
-                httpManager.AddManagedIdentityMockHandler(
-                    endpoint,
-                    Resource,
-                    MockHelpers.GetMsiSuccessfulResponse(),
-                    ManagedIdentitySourceType.CloudShell);
-
-                var result = await cca.AcquireTokenForClient(new string[] { scope })
-                    .WithManagedIdentity()
-                    .ExecuteAsync().ConfigureAwait(false);
-
-                Assert.IsNotNull(result);
-                Assert.IsNotNull(result.AccessToken);
-                Assert.AreEqual(TokenSource.IdentityProvider, result.AuthenticationResultMetadata.TokenSource);
-
-                result = await cca.AcquireTokenForClient(new string[] { scope })
-                    .WithManagedIdentity()
-                    .ExecuteAsync().ConfigureAwait(false);
-
-                Assert.IsNotNull(result);
-                Assert.AreEqual(TokenSource.Cache, result.AuthenticationResultMetadata.TokenSource);
-            }
-        }
-
-        [DataTestMethod]
-        [DataRow(TestConstants.ClientId)]
-        [DataRow("resourceId")]
-        public async Task CloudShellUserAssignedManagedIdentityNotSupportedAsync(string userAssignedClientId)
-        {
-            using (new EnvVariableContext())
-            using (var httpManager = new MockHttpManager())
-            {
-                SetEnvironmentVariables(Endpoint);
-
-                IConfidentialClientApplication cca = ConfidentialClientApplicationBuilder
-                    .Create("clientId")
-                    .WithHttpManager(httpManager)
-                    .WithExperimentalFeatures()
-                    .Build();
+                IManagedIdentityApplication mi = ManagedIdentityTests.CreateManagedIdentityApplication(httpManager, userAssignedClientId, userAssignedIdentityId);
```

Comment:
Please move `UserAssignedIdentityId` and other test helpers to their own files or at least 1 file clearly mentioning MI.